bids/modeling/statsmodels.py

```diff
@@ -659,8 +673,12 @@ def _collections_to_dfs(self, collections):
             # apply transformations
             transformations = self.node.transformations
             if transformations:
-                transformer = tm.TransformerManager(transformations['transformer'])
+                transformer = tm.TransformerManager(
+                    transformations['transformer'], keep_history=collection_history)
                 coll = transformer.transform(coll.clone(), transformations['instructions'])
+                
+                if hasattr(transformer, 'history_'):
+                    coll_hist[level] = transformer.history_
```

Comment:
`coll_hist` is a dictionary where keys are the level of the transformation. 

The logic in `_collections_to_dfs` applies transformations by level and then merges them.

Given that we want to keep the order of the collections as they go through Transformations this doesn't make sense for history.

That said, when would a Node have more than one level? Is it that Node of level `dataset` could get inputs from various levels? And those variable are transformed separately?